source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,241 @@
+following example instructs the driver to use ``users`` as the authentication
+database:
+
+.. code-block:: go
+   :copyable: true
+
+   uri := "mongodb://<username>:<password>@<hostname>:<port>/?authSource=users"
+   client := mongo.Connect(uri)
+
+Error Sending Message
+~~~~~~~~~~~~~~~~~~~~~
+
+When the driver fails to send a command after you make a request,
+it often displays the following general error message:
+
+.. code-block:: none
+   :copyable: false
+
+   com.mongodb.MongoSocketWriteException: Exception sending message
+
+The following sections describe methods that may help resolve the issue.
+
+Check Connection String
+-----------------------
+
+Verify that the connection string in your app is accurate. For more information
+about verifying your connection string, see
+:ref:`Connection Error <golang-troubleshooting-connection-string-port>`
+and :ref:`Authentication Error <golang-troubleshooting-connection-string-auth>`.
+
+Verify the Authentication Mechanism
+-----------------------------------
+
+Make sure you are using the correct authentication mechanism and credentials.
+For more information about authentication errors, see
+:ref:`Authentication Error <golang-troubleshooting-connection-auth-mechanism>`.
+
+Verify User Is in Authentication Database
+-----------------------------------------
+
+Verify the user is in the correct authentication database. For more
+information about the authentication database, see
+:ref:`Authentication Error <golang-troubleshooting-connection-admin>`.
+
+Configure Firewall
+------------------
+
+The firewall needs to have an open port for communicating with the MongoDB
+instance. For more information about configuring the firewall, see
+:ref:`Connection Error <golang-troubleshooting-connection-firewall>`.
+
+.. _golang-troubleshooting-connection-number-connections:
+
+Check the Number of Connections
+-------------------------------
+
+Each ``MongoClient`` instance supports a maximum number of concurrent open
+connections in its connection pool. The configuration parameter ``maxPoolSize``
+defines this value and is set to ``100`` by default. If there are already a
+number of open connections equal to ``maxPoolSize``, the server waits until
+a connection becomes available. If this wait time exceeds the ``maxIdleTimeMS``
+value, the driver responds with an error. For more information about how
+connection pooling works, see
+:ref:`How Does Connection Pooling Work in the Go Driver? <golang-faq-connection-pool>`
+in the FAQ.
+
+Timeout Error
+~~~~~~~~~~~~~
+
+Sometimes when you send a request through the driver to the server, the request
+times out. When this happens, you might receive an error message
+similar to the following message:
+
+.. code-block:: none
+   :copyable: false
+
+   timed out while checking out a connection from connection pool: context canceled
+
+If you receive this error, try the following methods to resolve the
+issue.
+
+Set ``Timeout`` Option
```

Comment:
S: remove monospace per the [style guide](https://www.mongodb.com/docs/meta/style-guide/style/titles-and-headings/#:~:text=Don%27t%20apply%20font%20treatments%20(bold%2C%20italics%2C%20or%20monospace)%20to%20text%20in%20a%20title%20or%20heading.)